loopy/tools.py

```diff
@@ -862,4 +865,38 @@ def t_unit_to_python(t_unit, var_name="t_unit",
     else:
         return python_code
 
+
+def memoize_on_disk(transformation_func, key_builder_t=LoopyKeyBuilder):
```

Comment:
Could you use this for `preprocess` and `buffer_array`? They have handcoded versions of the same logic.

Reply:
Sure, done!